google-cloud-storage/src/test/java/com/google/cloud/storage/conformance/retry/TestRetryConformance.java

```diff
@@ -0,0 +1,202 @@
+      InstructionList instruction,
+      boolean preconditionsProvided,
+      boolean expectSuccess,
+      int mappingId) {
+    this.host = host;
+    this.scenarioId = scenarioId;
+    this.method = requireNonNull(method, "method must be non null");
+    this.instruction = requireNonNull(instruction, "instruction must be non null");
+    this.preconditionsProvided = preconditionsProvided;
+    this.expectSuccess = expectSuccess;
+    this.mappingId = mappingId;
+    String instructionsString =
+        this.instruction.getInstructionsList().stream()
+            .map(s -> s.replace("return-", ""))
+            .collect(Collectors.joining("_"));
+    this.bucketName =
+        String.format("%s_bkt1_s%03d-%s-m%03d", BASE_ID, scenarioId, instructionsString, mappingId);
+    this.bucketName2 =
+        String.format("%s_bkt2_s%03d-%s-m%03d", BASE_ID, scenarioId, instructionsString, mappingId);
+    this.userProject =
+        String.format("%s_prj1_s%03d-%s-m%03d", BASE_ID, scenarioId, instructionsString, mappingId);
+    this.objectName =
+        String.format("%s_obj1_s%03d-%s-m%03d", BASE_ID, scenarioId, instructionsString, mappingId);
+  }
+
+  public String getHost() {
+    return host;
+  }
+
+  public String getBucketName() {
+    return bucketName;
+  }
+
+  public String getBucketName2() {
+    return bucketName2;
+  }
+
+  public String getUserProject() {
+    return userProject;
+  }
+
+  public String getObjectName() {
+    return objectName;
+  }
+
+  public byte[] getHelloWorldUtf8Bytes() {
+    return helloWorldUtf8Bytes;
+  }
+
+  public Path getHelloWorldFilePath() {
+    return helloWorldFilePath;
+  }
+
+  public int getScenarioId() {
+    return scenarioId;
+  }
+
+  public Method getMethod() {
+    return method;
+  }
+
+  public InstructionList getInstruction() {
+    return instruction;
+  }
+
+  public boolean isPreconditionsProvided() {
+    return preconditionsProvided;
+  }
+
+  public boolean isExpectSuccess() {
+    return expectSuccess;
+  }
+
+  public int getMappingId() {
+    return mappingId;
+  }
+
+  public ServiceAccountSigner getServiceAccountSigner() {
+    return serviceAccountCredentials;
+  }
+
+  public String getTestName() {
+    String instructionsDesc = Joiner.on("_").join(instruction.getInstructionsList());
+    return String.format(
+        "TestRetryConformance/%d-[%s]-%s-%d",
+        scenarioId, instructionsDesc, method.getName(), mappingId);
+  }
+
+  @Override
+  public String toString() {
+    return getTestName();
+  }
+
+  private static Path resolvePathForResource() {
+    ClassLoader cl = Thread.currentThread().getContextClassLoader();
+    URL url = cl.getResource("com/google/cloud/storage/conformance/retry/hello-world.txt");
+    assertNotNull(url);
+    try {
+      return Paths.get(url.toURI());
+    } catch (URISyntaxException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private static ServiceAccountCredentials resolveServiceAccountCredentials() {
```

Comment:
Why is this needed? Do you have some WithoutAuth type option for your client constructor?

Reply:
One of the code paths which can be used to sign things is only able to be used when it has a `com.google.auth.ServiceAccountSigner` `ServiceAccountCredentials`implements the Signer interface and is fairly easy to construct as long as we have a json (I'm reusing the one from the signing tests). This will only have an impact on the case we need to sign a url before uploading to it, otherwise the storage client is configured with `NoCredentials`.

Reply:
Interesting-- does the emulator even have support for signed URL uploads in any meaningful way? Is it possible to just have the test give a fake signed URL/signature to upload to?

Reply:
I'm not totally sure, admittedly my understanding of signed urls is still ramping up so I'm not sure if we are able to use a fake signed url. Something we'll have to look into.

Reply:
It does not support signed URLs.

Reply:
Do you know if that also extends to accepting a signed url to start a resumable upload?